webapi/src/main/scala/org/knora/webapi/slice/admin/api/PermissionsEndpoints.scala

```diff
@@ -0,0 +1,101 @@
+import zio.ZLayer
+
+import org.knora.webapi.messages.admin.responder.permissionsmessages.AdministrativePermissionCreateResponseADM
+import org.knora.webapi.messages.admin.responder.permissionsmessages.AdministrativePermissionGetResponseADM
+import org.knora.webapi.messages.admin.responder.permissionsmessages.AdministrativePermissionsForProjectGetResponseADM
+import org.knora.webapi.messages.admin.responder.permissionsmessages.ChangePermissionGroupApiRequestADM
+import org.knora.webapi.messages.admin.responder.permissionsmessages.ChangePermissionHasPermissionsApiRequestADM
+import org.knora.webapi.messages.admin.responder.permissionsmessages.ChangePermissionPropertyApiRequestADM
+import org.knora.webapi.messages.admin.responder.permissionsmessages.ChangePermissionResourceClassApiRequestADM
+import org.knora.webapi.messages.admin.responder.permissionsmessages.CreateAdministrativePermissionAPIRequestADM
+import org.knora.webapi.messages.admin.responder.permissionsmessages.CreateDefaultObjectAccessPermissionAPIRequestADM
+import org.knora.webapi.messages.admin.responder.permissionsmessages.DefaultObjectAccessPermissionCreateResponseADM
+import org.knora.webapi.messages.admin.responder.permissionsmessages.DefaultObjectAccessPermissionsForProjectGetResponseADM
+import org.knora.webapi.messages.admin.responder.permissionsmessages.PermissionDeleteResponseADM
+import org.knora.webapi.messages.admin.responder.permissionsmessages.PermissionGetResponseADM
+import org.knora.webapi.messages.admin.responder.permissionsmessages.PermissionsADMJsonProtocol
+import org.knora.webapi.messages.admin.responder.permissionsmessages.PermissionsForProjectGetResponseADM
+import org.knora.webapi.slice.admin.api.AdminPathVariables.groupIri
+import org.knora.webapi.slice.admin.api.AdminPathVariables.permissionIri
+import org.knora.webapi.slice.admin.api.AdminPathVariables.projectIri
+import org.knora.webapi.slice.common.api.BaseEndpoints
+
+final case class PermissionsEndpoints(base: BaseEndpoints) extends PermissionsADMJsonProtocol {
+
+  private val permissionsBase = "admin" / "permissions"
+
+  val postPermissionsAp = base.securedEndpoint.post
+    .in(permissionsBase / "ap")
+    .description("Create a new administrative permission")
+    .in(sprayJsonBody[CreateAdministrativePermissionAPIRequestADM])
+    .out(sprayJsonBody[AdministrativePermissionCreateResponseADM])
+
+  val getPermissionsApByProjectIri = base.securedEndpoint.get
+    .in(permissionsBase / "ap" / projectIri)
+    .description("Get all administrative permissions for a project.")
+    .out(sprayJsonBody[AdministrativePermissionsForProjectGetResponseADM])
+
+  val getPermissionsApByProjectAndGroupIri = base.securedEndpoint.get
+    .in(permissionsBase / "ap" / projectIri / groupIri)
+    .description("Get all administrative permissions for a project and a group.")
+    .out(sprayJsonBody[AdministrativePermissionGetResponseADM])
+
+  val getPermissionsDaopByProjectIri = base.securedEndpoint.get
```

Comment:
What `daop` means? This is probably subject to adjust for later, but I think it's wrongly abbreviated.

Reply:
You are right: Should be `doap` for `DefaultObjectAccessPermission` the same as the path segment this endpoint contains.

Fixed in 0f54648eff136381f4ec039a4f82f134c41bb59d